services/repository/src/main/java/org/sagebionetworks/repo/web/service/EntityServiceImpl.java

```diff
@@ -276,14 +272,13 @@ private void fireAfterCreateEntityEvent(UserInfo userInfo, Entity entity, Entity
 	 * @throws InvalidModelException
 	 */
 	private void fireAfterUpdateEntityEvent(UserInfo userInfo, Entity entity, EntityType type, boolean wasNewVersionCreated) throws NotFoundException, DatastoreException, UnauthorizedException, InvalidModelException{
-		List<EntityProvider<? extends Entity>> providers = metadataProviderFactory.getMetadataProvider(type);
-		if(providers != null) {
-			for (EntityProvider<? extends Entity> provider : providers) {
-				if (provider instanceof TypeSpecificUpdateProvider) {
-					((TypeSpecificUpdateProvider) provider).entityUpdated(userInfo, entity, wasNewVersionCreated);
-				}
+		Optional<EntityProvider<? extends Entity>> provider = metadataProviderFactory.getMetadataProvider(type);
```

Comment:
The provider local variable is not actually needed.  All cases can be shorted to:
`metadataProviderFactory.getMetadataProvider(type).ifPresent(type->{
....
})`